### PR TITLE
Olm bundle

### PR DESCRIPTION
**Pull request by locriandev**
Ref. [ART-3676](https://issues.redhat.com/browse/ART-3676)

It turned out that some refactoring was enough. This PR cleans up the code:
- EXTRAS_ADVISORY and METADATA_ADVISORY params are never used
- Elliott is no longer needed
- Only a single Doozer invocation is left for olm-bundle:rebase-and-build; therefore, expanding pyartcd with a new pipeline seemed like an overkill
- The groovy library file has been removed, what was left in it has been moved to the 'Build bundles' stage within the Jenkinsfile

The job has been tested in a hack space: https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/hack/job/dpaolell/job/olm_bundle/29/console

### CODE REVIEW

File: jobs/build/olm_bundle/Jenkinsfile [hunk 1]
```diff
@@ -73,20 +72,6 @@ pipeline {
             defaultValue: '',
             trim: true,
         )
-        string(
```
Comment (thegreyd) [inline jobs/build/olm_bundle/Jenkinsfile:76]: Can you check the jobs readme, and remove these params from there as well?

File: jobs/build/olm_bundle/Jenkinsfile [hunk 2]
```diff
@@ -97,54 +82,31 @@ pipeline {
...
-                    } else {
-                        currentBuild.displayName  += " ${params.EXTRAS_ADVISORY ?: ''}"
-                    }
-                    if (operator_nvrs && (only || exclude)) {
-                        error("Cannot use OPERATOR_NVRS with ONLY or EXCLUDE.")
-                    }
-                    if (operator_nvrs && params.EXTRAS_ADVISORY) {
-                        error("Cannot use OPERATOR_NVRS with EXTRAS_ADVISORY.")
-                    }
-                    olm_bundles.buildlib.initialize(false, false)  // ensure kinit
```
Comment (thegreyd) [inline jobs/build/olm_bundle/Jenkinsfile:125]: Don't see this as well, not needed?

Comment (locriandev) [reply jobs/build/olm_bundle/Jenkinsfile:125]: With EXTRAS_ADVISORY param being removed, this branch will never be (and currently never is) entered

File: jobs/build/olm_bundle/Jenkinsfile [hunk 3]
```diff
@@ -97,54 +82,31 @@ pipeline {
...
-
-                    if (params.ASSEMBLY && params.ASSEMBLY != "stream") {
-                        currentBuild.displayName += " - assembly ${params.ASSEMBLY}"
-                        if (params.EXTRAS_ADVISORY || params.METADATA_ADVISORY) {
-                            error("Cannot use EXTRAS_ADVISORY or METADATA_ADVISORY when building for non-stream assembly.")
-                        }
-                    } else {
-                        currentBuild.displayName  += " ${params.EXTRAS_ADVISORY ?: ''}"
-                    }
-                    if (operator_nvrs && (only || exclude)) {
```
[the remark this answers is not in the document]
Comment (locriandev) [reply jobs/build/olm_bundle/Jenkinsfile:119]: With EXTRAS_ADVISORY being removed, this is dead code

[the remark this answers is not in the document]
Comment (thegreyd) [reply jobs/build/olm_bundle/Jenkinsfile:119]: So we can remove only and exclude params as well

[the remark this answers is not in the document]
Comment (locriandev) [reply jobs/build/olm_bundle/Jenkinsfile:119]: I think ONLY and EXCLUDE can be useful for testing purposes, even if not triggered by automation. I'd leave them be

**Pull request merged**